internal/data/data.go

```diff
@@ -0,0 +1,102 @@
+package data
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/viper"
+	"gopkg.in/yaml.v3"
```

Comment:
I reflexively changed this to a more widely used YAML library, but after looking up the one you were using (github.com/goccy/go-yaml) that looks perfectly fine too.